src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -875,7 +875,7 @@ private function syncReferences($referencesToUpdate)
             foreach ($toUpdate as $nodeId => $references) {
                 foreach ($references['properties'] as $name => $data) {
                     foreach ($data['values'] as $value) {
-                        $targetId = $this->getSystemIdForNodeUuid($value);
+                        $targetId = $this->getSystemIdForNode($value);
```

Comment:
when we create the a node we are referencing to in the same session ``$value`` is the path and not the UUID causing the below exception to be triggered.